.github/workflows/emscripten.yml

```diff
@@ -36,86 +33,36 @@ concurrency:
 
 permissions:
-  contents: read # to fetch code (actions/checkout)
+  contents: read  # to fetch code (actions/checkout)
 
 jobs:
   build-wasm-emscripten:
     name: Build NumPy distribution for Pyodide
     runs-on: ubuntu-22.04
     # To enable this workflow on a fork, comment out:
     if: github.repository == 'numpy/numpy'
-    env:
-      PYODIDE_VERSION: 0.26.0
-      # PYTHON_VERSION and EMSCRIPTEN_VERSION are determined by PYODIDE_VERSION.
-      # The appropriate versions can be found in the Pyodide repodata.json
-      # "info" field, or in Makefile.envs:
-      # https://github.com/pyodide/pyodide/blob/main/Makefile.envs#L2
-      PYTHON_VERSION: 3.12.1
-      EMSCRIPTEN_VERSION: 3.1.58
-      NODE_VERSION: 18
     steps:
       - name: Checkout NumPy
-        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
         with:
           submodules: recursive
-          # This input shall fetch tags without the need to fetch the
-          # entire VCS history, see https://github.com/actions/checkout#usage
           fetch-tags: true
 
-      - name: Set up Python ${{ env.PYTHON_VERSION }}
+      - name: Set up Python
         id: setup-python
-        uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
+        uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d  # v5.1.0
         with:
-          python-version: ${{ env.PYTHON_VERSION }}
+          python-version: "3.12.1"
 
-      - name: Set up Emscripten toolchain
-        uses: mymindstorm/setup-emsdk@6ab9eb1bda2574c4ddb79809fc9247783eaf9021 # v14
-        with:
-          version: ${{ env.EMSCRIPTEN_VERSION }}
-          actions-cache-folder: emsdk-cache
-
-      - name: Install pyodide-build
-        run: pip install pyodide-build==${{ env.PYODIDE_VERSION }}
-
-      - name: Find installation for pyodide-build
-        shell: python
-        run: |
-          import os
-          import pyodide_build
-          from pathlib import Path
-
-          pyodide_build_path = Path(pyodide_build.__file__).parent
-
-          env_file = os.getenv('GITHUB_ENV')
+      - uses: pypa/cibuildwheel@a8d190a111314a07eb5116036c4b3fb26a4e3162  # v2.19.0
+        env:
+          CIBW_PLATFORM: pyodide
 
-          with open(env_file, "a") as myfile:
-              myfile.write(f"PYODIDE_BUILD_PATH={pyodide_build_path}\n")
-
-      - name: Build NumPy for Pyodide
-        run: |
-          pyodide build \
-          -Cbuild-dir=build \
-          -Csetup-args="--cross-file=$PWD/tools/ci/emscripten/emscripten.meson.cross" \
-          -Csetup-args="-Dblas=none" \
-          -Csetup-args="-Dlapack=none"
-
-      - name: Set up Node.js
-        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+      - name: Upload wheel artifact(s)
+        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808  # v4.3.3
```

Comment:
I see that other workflows upload artifacts on `failure()` events, while upload them without conditions. I have added this for the sake of completeness/convenience, but it's not needed if we don't plan to share it across jobs, at least for now (the documentation builds can be adjusted to need them later). Happy to keep this or remove this based on further directions.